### PR TITLE
Simplify the implementation of sbrk().

### PR DESCRIPTION
**Pull request by juj**
Reading the implementation of sbrk(), it has evolved to be a bit #ifdef heavy. This cleans it up a little, and avoids the use of goto.

### CODE REVIEW

File: system/lib/libc/sbrk.c
Comment (kripken) [inline system/lib/libc/sbrk.c:99]: Why is this check now outside of the ifdef for shared memory? I guess the optimizer can remove it in the single-threaded case but it would be simpler to leave it in the ifdef, I think?

Comment (juj) [reply system/lib/libc/sbrk.c:99]: The idea was to avoid most of the ifdefs inside the inner loop for simpler readability, and indeed leave it to the optimizer to remove.

Though now the CI is saying that it is not optimizing something out since wasm2js code size test increased (but oddly only wasm2js code sizes..)

Comment (juj) [reply system/lib/libc/sbrk.c:99]: Oh I see, re-reading the code, it's definitely simpler to move it up inside that one remaining shared memory ifdef.. I'll update to that.

File: system/lib/libc/sbrk.c
Comment (kripken) [inline system/lib/libc/sbrk.c:90]:
```suggestion
#ifdef __EMSCRIPTEN_SHARED_MEMORY__
    uintptr_t expected = old_brk;
```

Comment (kripken) [reply system/lib/libc/sbrk.c:90]: (otherwise it is unused in single-threaded and I expect the compiler will warn/error)

File: system/lib/libc/sbrk.c
Comment (sbc100) [inline system/lib/libc/sbrk.c:64]: I don't think we need to make this distinction because `__c11_atomic_load` works fine, even on single-threads builds (it will just lower to a normal load).

We might even be able o avoid the `__c11_atomic_load` completely if we just give it the correct type since reads from `_Atomic` variables lower to `__c11_atomic_load`.

Comment (juj) [reply system/lib/libc/sbrk.c:64]: Sure, updated

File: system/lib/libc/sbrk.c
Comment (juj) [reply system/lib/libc/sbrk.c:90]: Makes sense, updated.

File: system/lib/libc/sbrk.c
Comment (juj) [reply system/lib/libc/sbrk.c]: Sure, updated.

File: system/lib/libc/sbrk.c
Comment (juj) [reply system/lib/libc/sbrk.c]: > even on single-threads builds (it will just lower to a normal load).
Turns out that is not correct. It will cause an error
```
wasm-ld: error: --shared-memory is disallowed by sbrk.o because it was not compiled with 'atomics' or 'bulk-memory' features.
```

Comment (sbc100) [reply system/lib/libc/sbrk.c:64]: Hmm.. we should be able to fix that.  I'll take a look after this lands.

Comment (sbc100) [reply system/lib/libc/sbrk.c:64]: We can take care of this one we address #20803.  Resolve this here.

**Review by sbc100 — APPROVED** [review]

**Pull request merged**
